github3/github.py

```diff
@@ -57,7 +58,8 @@ class GitHub(GitHubCore):
     """
 
     def __init__(self, username='', password='', token=''):
-        super(GitHub, self).__init__({})
+        session = GitHubSession()
```

Comment:
One thing that was convenient in the past was not having calls to `GitHubSession` sprinkled everywhere. What do you think of having a `GitHubCore` method `new_session` that returns a new `GitHubSession` so we can do

```
super(GitHub, self).__init__({}, self.new_session())
```

And likewise with other things like GitHubStatus.

Reply:
Ah, that could work pretty well. Mostly I was trying to avoid the end user from having to instantiate their own session first, so that they could continue using `foo = GitHub()` as they do today. Your way should still support that, I'll play around with it.

Reply:
Yeah, I agree that the user shouldn't have to instantiate it themselves.

Reply:
First pass of trying to do it this way resulted in a recursion depth issue.